src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -520,6 +538,22 @@ private BoundExpression TransformCompoundAssignmentLHS(BoundExpression originalL
                     }
                     break;
 
+                case BoundKind.IndexOrRangePatternIndexerAccess:
+                    {
+                        var patternIndexerAccess = (BoundIndexOrRangePatternIndexerAccess)originalLHS;
+                        (RefKind refKind, bool isRange) = patternIndexerAccess.PatternSymbol switch
```

Comment:
I might be missing something but it doesn't look like `isRange` is used here.

Reply:
You're right, I managed to simplify the code.